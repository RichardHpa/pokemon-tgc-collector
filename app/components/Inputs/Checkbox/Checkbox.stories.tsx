import { Checkbox as CheckboxComponent } from "./Checkbox";

import type { ComponentStory, ComponentMeta } from "@storybook/react";

const Template: ComponentStory<typeof CheckboxComponent> = ({ ...args }) => {
  return (
    <CheckboxComponent
      {...args}
      label="Input Field Label"
      name="storybook-checkbox"
    />
  );
};

export const Checkbox = Template.bind({});

export default {
  title: "Components/Inputs/Checkbox",
  component: CheckboxComponent,
} as ComponentMeta<typeof CheckboxComponent>;
