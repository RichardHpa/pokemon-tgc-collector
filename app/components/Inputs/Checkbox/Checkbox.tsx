import { forwardRef } from "react";

import type { InputFieldProps } from "../InputField/types";

export const Checkbox = forwardRef<
  HTMLInputElement,
  Omit<InputFieldProps, "type">
>(
  (
    { disabled = false, error, label, name, required = false, ...props },
    ref
  ) => {
    return (
      <div>
        <div className="flex items-center">
          <input
            {...props}
            ref={ref}
            id={name}
            name={name}
            type="checkbox"
            disabled={disabled}
            required={required}
            aria-invalid={error ? true : undefined}
            aria-describedby={`${name}-error`}
            className="h-4 w-4 rounded border-gray-300 text-blue-600 focus:ring-blue-500"
          />
          <label
            htmlFor={name}
            className="ml-2 block cursor-pointer text-sm text-gray-900"
          >
            {label} {required && <span>*</span>}
          </label>
        </div>
        {error && (
          <div className="pt-1 text-red-700" id={`${name}-error`}>
            {error}
          </div>
        )}
      </div>
    );
  }
);

Checkbox.displayName = "Checkbox";
