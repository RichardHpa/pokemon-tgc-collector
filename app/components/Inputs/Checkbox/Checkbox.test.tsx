import { render, screen } from "@testing-library/react";

import { withMarkup } from "test/helpers/withMarkup";

import { Checkbox } from "./Checkbox";

const checkboxProps = {
  label: "checkbox",
  name: "checkbox",
};

describe("Checkbox", () => {
  test("renders Checkbox", () => {
    render(<Checkbox {...checkboxProps} />);
    expect(screen.getByRole("checkbox")).toBeInTheDocument();
  });

  test("renders Checkbox with label", () => {
    render(<Checkbox {...checkboxProps} />);
    expect(screen.getByText("checkbox")).toBeInTheDocument();
  });

  test("renders Checkbox with required", () => {
    const { getByText } = render(<Checkbox {...checkboxProps} required />);
    const getByTextWithMarkup = withMarkup(getByText);
    getByTextWithMarkup("checkbox *");
  });

  test("renders Checkbox with error", () => {
    render(<Checkbox {...checkboxProps} error="error" />);
    expect(screen.getByText("error")).toBeInTheDocument();
  });

  test("renders Checkbox snapshot", () => {
    const { container } = render(<Checkbox {...checkboxProps} />);
    expect(container).toMatchSnapshot();
  });
});
