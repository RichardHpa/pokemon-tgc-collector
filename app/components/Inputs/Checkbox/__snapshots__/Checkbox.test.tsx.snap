// Vitest Snapshot v1

exports[`Checkbox > renders Checkbox snapshot 1`] = `
<div>
  <div>
    <div
      class="flex items-center"
    >
      <input
        aria-describedby="checkbox-error"
        class="h-4 w-4 rounded border-gray-300 text-blue-600 focus:ring-blue-500"
        id="checkbox"
        name="checkbox"
        type="checkbox"
      />
      <label
        class="ml-2 block cursor-pointer text-sm text-gray-900"
        for="checkbox"
      >
        checkbox
         
      </label>
    </div>
  </div>
</div>
`;
